client/cypress/e2e/earn.cy.ts

```diff
@@ -184,5 +187,55 @@ Object.values(viewports).forEach(({ device, viewportWidth, viewportHeight, isDes
             .should('have.text', '1 GLM');
         });
     });
+
+    it('Wallet connected: Lock 1000 GLM + move epoch', () => {
+      connectWallet();
+
+      cy.get('[data-test=BoxGlmLock__Section--current__DoubleValue__primary]')
+        .invoke('text')
+        .then(text => {
+          const amountToLock = 1000;
+          const lockedGlms = parseInt(text.replace(/\u200a/g, ''), 10);
+
+          cy.get('[data-test=BoxGlmLock__Button]').click();
+          cy.get('[data-test=InputsCryptoFiat__InputText--crypto]').clear().type(`${amountToLock}`);
+          cy.get('[data-test=GlmLockTabs__Button]').should('have.text', 'Lock');
+          cy.get('[data-test=GlmLockTabs__Button]').click();
+          cy.get('[data-test=GlmLockTabs__Button]').should('have.text', 'Waiting for confirmation');
+          cy.confirmMetamaskPermissionToSpend({
+            spendLimit: '99999999999999999999',
+          });
+          cy.get('[data-test=GlmLockTabs__Button]', { timeout: 180000 }).should(
+            'have.text',
+            'Close',
+          );
+          cy.get('[data-test=GlmLockNotification--success]').should('be.visible');
+          cy.get('[data-test=GlmLockTabs__Button]').click();
+          cy.wait(5000);
+          cy.window().then(async win => {
+            await win.mutateAsyncMoveEpoch();
+            cy.wait(5000);
+            await axios.post(`${env.serverEndpoint}snapshots/pending`);
+            cy.wait(5000);
+            cy.reload();
```

Comment:
Please extract that to an util and add a comment there explaining why calling `snapshots/pending` is required, why waiting is required and why reload is required.